File: tests/src/main/java/com/abstratt/simon/examples/UI3.java
package com.abstratt.simon.examples;

import java.util.List;

import com.abstratt.simon.examples.UI.Application;
import com.abstratt.simon.metamodel.dsl.Meta;

@Meta.Package(builtIns = {})
public interface UI3 {
	interface IPrototype {
		@Meta.Reference
		@Meta.Typed(Application.class)
		List<Application> applications();
	}
}

File: example-languages-kotlin/src/main/kotlin/com/abstratt/simon/examples/kotlin/UI3.kt
package com.abstratt.simon.examples.kotlin

import com.abstratt.simon.metamodel.dsl.Meta

@Meta.Package(builtIns = [])
interface UI3 {
    interface IPrototype {
        @Meta.Reference
        @Meta.Typed(UI.Application::class)
        fun applications(): List<UI.Application?>?
    }
}